src/evaluate/evaluate.ts

```diff
@@ -192,13 +192,12 @@ export class Evaluator {
       ev(x.props),
     ]) as CfnResource;
 
-    if (x.creationPolicy) {
-      resource.cfnOptions.creationPolicy = x.creationPolicy;
-    }
-
-    if (x.updatePolicy) {
-      resource.cfnOptions.updatePolicy = x.updatePolicy;
-    }
+    resource.cfnOptions.creationPolicy = x.creationPolicy;
+    resource.cfnOptions.updatePolicy = x.updatePolicy;
+    resource.cfnOptions.metadata = x.metadata;
+    resource.cfnOptions.updateReplacePolicy =
+      x.updateReplacePolicy as CfnDeletionPolicy;
+    resource.cfnOptions.deletionPolicy = x.deletionPolicy as CfnDeletionPolicy;
```

Comment:
Should this also be supported by Constructs via `applyRemovalPolicy()`

Reply:
Same thing as for metadata.